service/orchestrator/cloud_service.go

```diff
@@ -134,32 +132,32 @@ func (s *Service) RemoveCloudService(_ context.Context, req *orchestrator.Remove
 	return &emptypb.Empty{}, nil
 }
 
-// CreateDefaultTargetCloudService creates a new "default" target cloud services,
-// if no target service exists in the database.
-//
-// If a new target cloud service was created, it will be returned.
-func (s *Service) CreateDefaultTargetCloudService() (service *orchestrator.CloudService, err error) {
-	var count int64
-	s.db.Model(&orchestrator.CloudService{}).Count(&count)
-
-	if count == 0 {
-		// Create a default target cloud service
-		service =
+// CreateDefaultTargetCloudService implements method for OrchestratorServer interface for creating a new "default"
+// target cloud services, if no target service exists in the database.
+// Returns new created target cloud service. Otherwise, returns nil (if a record exists) or error if sth went wrong
+func (s *Service) CreateDefaultTargetCloudService() (*orchestrator.CloudService, error) {
+	var storedServices []*orchestrator.CloudService
+	err := s.db.Read(&storedServices)
```

Comment:
Same as above, please provide a Count interface in the persistence layer. this is unnecessarily resource exhaustive.